src/lando/api/tests/test_landings.py

```diff
@@ -565,7 +563,17 @@ def test_merge_conflict(
 
     assert worker.run_job(job)
     assert job.status == LandingJobStatus.FAILED
-    assert "hunks FAILED" in caplog.text
+
+    # Make sure a conflict actually happened.
+    assert repo_type in [
+        SCM_TYPE_GIT,
+        SCM_TYPE_HG,
+    ], f"Unsupported repository type: {repo_type=}"
+    if repo_type == SCM_TYPE_GIT:
+        assert "Rejected hunk" in caplog.text
+    elif repo_type == SCM_TYPE_HG:
+        assert "hunks FAILED" in caplog.text
+
```

Comment:
You should add an extra argument to this test's parameters which is the expected string to search for when a conflict is detected. That way you can remove the "Unsupported repository type" assertion and instead do `assert scm_conflict_string in caplog.text`.

Reply:
Agreed with this, also we don't need the first assert here, since `repo_type` is explicitly set in the parameters.

Reply:
Ah, yep, nice!